src/v/cluster/topic_updates_dispatcher.cc

```diff
@@ -410,12 +412,37 @@ topic_updates_dispatcher::collect_in_progress(
     in_progress.reserve(current_assignments.size());
     // collect in progress assignments
     for (auto& p : current_assignments) {
-        auto previous = _topic_table.local().get_previous_replica_set(
-          model::ntp(tp_ns.ns, tp_ns.tp, p.id));
-        if (previous) {
-            in_progress.emplace(p.id, std::move(previous.value()));
+        model::ntp ntp(tp_ns.ns, tp_ns.tp, p.id);
+        const auto& in_progress_updates
+          = _topic_table.local().updates_in_progress();
+        auto it = in_progress_updates.find(ntp);
+        if (it == in_progress_updates.end()) {
+            continue;
+        }
+        const auto state = it->second.get_state();
+        if (state == reconfiguration_state::in_progress) {
+            auto previous = _topic_table.local().get_previous_replica_set(ntp);
+            if (previous) {
+                in_progress.emplace(p.id, std::move(previous.value()));
+            }
+        } else {
+            vassert(
+              state == reconfiguration_state::cancelled
+                || state == reconfiguration_state::force_cancelled,
+              "Invalid reconfiguration state: {}",
+              state);
+
+            auto target = _topic_table.local().get_target_replica_set(ntp);
+            if (target) {
+                in_progress.emplace(p.id, std::move(target.value()));
+            }
         }
     }
+    vlog(
+      clusterlog.trace,
+      "Deleting ntp: {},  in_progress ops: {}",
+      tp_ns,
+      in_progress);
```

Comment:
nit: can we move this log somewhere else, this method is not really deleting anything